src/inmanta/deploy/scheduler.py

```diff
@@ -651,37 +808,107 @@ async def report_resource_state(
             if deployment_result is not None:
                 # first update state, then send out events
                 self._deploying_latest.remove(resource)
+
+                previous_deployment_result = state.deployment_result
+
                 state.deployment_result = deployment_result
                 self._work.finished_deploy(resource)
+
+                # Set of dependant resources that are interested in deploys of the current resource
+                concerned_resources: set[ResourceIdStr] = set()
+                provides: Set[ResourceIdStr]
                 if deployment_result is DeploymentResult.DEPLOYED:
                     self._state.dirty.discard(resource)
+
+                    if previous_deployment_result != DeploymentResult.DEPLOYED:
```

Comment:
These changes are mostly unrelated to the rest, I think?

Reply:
I'm not sure anymore, this is what was proposed [here (last paragraph)](https://github.com/inmanta/inmanta-core/pull/8292#discussion_r1827968749).

Reply:
See @sanderr 's answer https://inmanta.slack.com/archives/CKRF0C8R3/p1732551457970899?thread_ts=1732551090.789439&cid=CKRF0C8R3

Reply:
I would add this explanation to the method definition. 

This method does a lot, and it has no docstring

Reply:
> it has no docstring

It does in the ABC